packages/Textarea/src/Textarea.js

```diff
@@ -8,97 +8,109 @@ const propTypes = {
   a11yText: PropTypes.string,
   canExpand: PropTypes.bool,
   className: PropTypes.string,
+  /** Do not use in conjunction with value prop */
+  defaultValue: PropTypes.string,
   hasError: PropTypes.bool,
   inputRef: PropTypes.func,
   isDisabled: PropTypes.bool,
   isReadOnly: PropTypes.bool,
   maxHeight: PropTypes.string,
+  onChange: PropTypes.func,
   size: PropTypes.oneOf(ShirtSizes.DEFAULT),
+  /** Do not use in conjunction with defaultValue prop */
   value: PropTypes.string,
 };
 
 const defaultProps = {
   a11yText: null,
   canExpand: true,
   className: null,
+  defaultValue: "",
   hasError: false,
   inputRef: () => {},
   isDisabled: false,
   isReadOnly: false,
   maxHeight: "300px",
+  onChange: () => {},
   size: ShirtSizes.MEDIUM,
-  value: "",
+  value: null,
 };
 
-class Textarea extends React.Component {
-  componentDidMount() {
-    if (this.props.canExpand) {
-      this.resize();
-      window.addEventListener("resize", this.resize);
+function Textarea(props) {
+  let textarea = null;
+
+  const resize = () => {
+    if (textarea && textarea.style) {
+      textarea.style.height = 0;
+      textarea.style.height = `${textarea.scrollHeight + 2}px`;
     }
-  }
+  };
 
-  componentDidUpdate(prevProps) {
-    if (this.props.canExpand && this.props.value !== prevProps.value) {
-      this.resize();
+  React.useEffect(() => {
+    if (props.canExpand) {
+      resize();
+      window.addEventListener("resize", resize);
     }
-  }
 
-  componentWillUnmount() {
-    window.removeEventListener("resize", this.resize);
-  }
+    return function cleanup() {
+      window.removeEventListener("resize", resize);
+    };
+  });
 
-  setRef = node => {
-    this.textarea = node;
-    this.props.inputRef(node);
-  };
-
-  resize = () => {
-    if (this.textarea && this.textarea.style) {
-      this.textarea.style.height = 0;
-      this.textarea.style.height = `${this.textarea.scrollHeight + 2}px`;
+  React.useEffect(() => {
+    if (props.canExpand) {
+      resize();
     }
+  }, [props.value]);
```

Comment:
`canExpand` is also a dependency of this effect.
Which actually indicates a slight issue with this component that it stops the data flow https://overreacted.io/writing-resilient-components/#dont-stop-the-data-flow-in-rendering
If I type a bunch of text in an expandable textarea and then pass in `canExpand={false}` it will not revert to an unexpanded look.